File: SecondYearCS/OOP/CivitasRB/lib/CasillaImpuesto.rb
require_relative 'sorpresa.rb'
require_relative 'casilla.rb'
require_relative 'tipo_sorpresa.rb'
require_relative 'titulo_propiedad.rb'
require_relative 'mazo_sorpresa.rb'
require_relative 'diario.rb'
require_relative 'dado.rb'
module CivitasRB
  class CasillaImpuesto < Casilla
    def initialize(cantidad, nombre)
      super(nombre)
      @importe = cantidad
    end

    #LLamar al metodo paga impuesto de la clase jugador
    def recibeJugador(act,todos)
      if self.jugadorCorrecto(act,todos)
        self.informe(act,todos)
        todos.at(act).pagaImpuesto(@importe)
      end
    end

   end
end

File: SecondYearCS/OOP/Civitas/src/civitas/CasillaImpuesto.java
package civitas;
import java.util.ArrayList;

public class CasillaImpuesto extends Casilla {

    private float importe;
    //Tipo Impuesto
    CasillaImpuesto(float cantidad, String nombre){
        super(nombre);
        this.importe = cantidad;
    }
    
    @Override
    void recibeJugador(int actual, ArrayList<Jugador> todos){
        if( this.jugadorCorrecto(actual, todos) ){
            this.informe(actual,todos);
            todos.get(actual).pagaImpuesto(this.importe);
        }
    }

    @Override
    public String toString(){
       String result = super.toString();
       return result;
    } 
}
